src/custom.d.ts

```diff
@@ -12,15 +12,15 @@ declare module '*.md';
 
 // Reference https://github.com/tshelburne/react-children-by-type
 declare module 'react-children-by-type' {
-  import type { ReactNode, JSXElementConstructor } from 'react';
+  import * as React from 'react';
 
   export function allByType<P>(
-    children: ReactNode,
-    type: JSXElementConstructor<P>,
-  ): ReactElement<P>[];
+    children: React.ReactNode,
+    type: React.JSXElementConstructor<P>,
+  ): React.ReactElement<P>[];
```

Comment:
Before these changes, calling `allByType` would return `React.ReactElement<any>[]`.

No idea why. Seems like it should work, and the types here are pretty straightforward.

Also no idea why changing to a star import "fixes" it... Tried for a while to figure it out.